scripts/validate_tinkerbell_hardware_file.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+E2E_BINARY="${1?Specify first argument - E2E tests binary for the current build}"
+TINKERBELL_HARDWARE_COUNT_FILE="${2?Specify second argument - tinkerbell hardware requirements file}"
+
+ALL_TINKERBELL_TESTS=$(exec $E2E_BINARY -test.list 'TestTinkerbell')
+
+declare -A TINKERBELL_HARDWARE_COUNT_LIST
+
+HARDWARE_COUNT_VALIDATION_STATUS=true 
+while IFS="@" read -r TEST_NAME HARDWARE_COUNT
+do
+    TINKERBELL_HARDWARE_COUNT_LIST[$TEST_NAME]="$HARDWARE_COUNT"
+done < <(yq e 'to_entries | .[] | (.key + "@" + .value)' ${TINKERBELL_HARDWARE_COUNT_FILE})
+for test in $ALL_TINKERBELL_TESTS
```

Comment:
To tie in with the Bash array change above, we can change the iteration to expand the array
```suggestion
for test in ${ALL_TINKERBELL_TESTS[@]}
```